prepare/metrics/hhem.py

```diff
@@ -0,0 +1,48 @@
+from unitxt import add_to_catalog
+from unitxt.metrics import FaithfulnessHHEM, MetricPipeline
+from unitxt.operators import Copy
+from unitxt.test_utils.metrics import test_metric
+
+pairs = [
+    ("The capital of France is Berlin.", "The capital of France is Paris."),
+    ("I am in California", "I am in United States."),
+    ("I am in United States", "I am in California."),
+]
+
+task_data = [{"contexts": [p[0]], "answer": p[1]} for p in pairs]
+
+metric = MetricPipeline(
+    main_score="score",
+    preprocess_steps=[
+        Copy(
+            field_to_field={
```

Comment:
Why do these copies of fields?

If I understand this is a reference less  a faithfulness metrics that should receive the model prediction and the contexts (from task_data).

Reply:
Just added a fix (that required a bigfix in operator as well), but the idea remains.
This metric needs to support two usecases:
1. regular flow: model prediction, and contexts from task_data 
2. running on external rag data: the input is a dict with the fields "contexts", "answer" 
This should capture both